### PR TITLE
metrics/influxdb: Support V2

### PR DESCRIPTION
**Pull request by jwasinger**
This PR builds on https://github.com/ethereum/go-ethereum/pull/22726:
    * adds flag to enable v2 (`--metrics.influxdbv2`), flags for v2-specific features (`--metrics.influxdb.token`, `--metrics.influxdb.bucket`), also carries over addition of support for specifying organization (`--metrics.influxdb.organization` from #22726).
    * retains backwards compatibility with influxdb-v1.

### CODE REVIEW

**Comment by jwasinger** [comment]
I can see that this seems to work by looking at the schemas created in Influx, and also seeing that Geth is successfully pushing data.  However, I ran into issues connecting it to Grafana using the auth token.  Tried adding the auth token via an http header as suggested in https://community.grafana.com/t/grafana-issue-connecting-to-influxdb-2-0-3-influxql/42757/4 .  But that didn't work.  And I probably won't have too much free time to continue digging on that this week.

File: metrics/influxdb/influxdbv2.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
+package influxdb
```
Comment (karalabe) [inline metrics/influxdb/influxdbv2.go:1]: PLs add file header

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -1744,11 +1767,17 @@ func SetupMetrics(ctx *cli.Context) {
 		log.Info("Enabling metrics collection")
 
 		var (
-			enableExport = ctx.GlobalBool(MetricsEnableInfluxDBFlag.Name)
-			endpoint     = ctx.GlobalString(MetricsInfluxDBEndpointFlag.Name)
-			database     = ctx.GlobalString(MetricsInfluxDBDatabaseFlag.Name)
-			username     = ctx.GlobalString(MetricsInfluxDBUsernameFlag.Name)
-			password     = ctx.GlobalString(MetricsInfluxDBPasswordFlag.Name)
+			enableExport   = ctx.GlobalBool(MetricsEnableInfluxDBFlag.Name)
```
Comment (karalabe) [inline cmd/utils/flags.go:1770]: A bit further above we could add 

```
CheckExclusive(ctx, MetricsEnableInfluxDBFlag, MetricsEnableInfluxDBV2Flag)
```

Comment (karalabe) [reply cmd/utils/flags.go:1770]: Maybe also add checks so if v1 is enabled, all the v1 flags are checked to be specified and non of the v2 ones and vice versa. Would avoid a lot of issues opened :D

File: cmd/utils/flags.go [hunk 3]
```diff
@@ -761,6 +761,29 @@ var (
...
 	}
 
+	MetricsEnableInfluxDBV2Flag = cli.BoolFlag{
+		Name:  "metrics.influxdbv2",
+		Usage: "Enable metrics export/push to an external InfluxDB v2 database",
+	}
+
+	MetricsInfluxDBTokenFlag = cli.StringFlag{
+		Name:  "metrics.influxdb.token",
+		Usage: "Token to authorize access to the database (v2 only)",
```
Comment (holiman) [inline cmd/utils/flags.go:771]: This one, and the three below, all say "(v2 only)" in the Usage portion. Maybe we should just change their namespace to be `metrics.invfluxdbv2.token` etc ?

**Review by holiman — COMMENTED** [review]
Looks pretty good to me!

File: cmd/utils/flags.go (hunk 3 above)
Comment (jwasinger) [reply cmd/utils/flags.go:771]: Okay.  Also noted on the descriptions of the username/password flags that those are v1-only.

**Comment by holiman** [comment]
Rebased on master so I can test it more easily

**Comment by holiman** [comment]
Ok, tested this with a local node against our grafana/influxv1 (obs: version 1) cluster. Seems to work fine, in the sense that it doesn't break the old stuff.

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**